Sources/NIOHPACK/HPACKHeader.swift

```diff
@@ -276,42 +265,159 @@ public struct HPACKHeaders: ExpressibleByDictionaryLiteral, Sendable {
     /// - Returns: A list of the values for that header field name.
     @inlinable
     public subscript(canonicalForm name: String) -> [String] {
-        let result = self[name]
-        guard result.count > 0 else {
+        return self._values(forHeader: name, canonicalForm: true)
+    }
+
+    /// Return a sequence of header values for the header field named `name`.
+    ///
+    /// See also ``subscript(name:)`` and ``subscript(canonicalForm:)``.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the header field whose values should be iterated.
+    ///   - canonicalForm: Whether the values should be decomposed into their "canonical form". That
+    ///     is, splitting them on commas as extensively as possible such that multiple values
+    ///     received on the one line are returned as separate entries.
+    /// - Note: Enabling `canonicalForm`  for the 'set-cookie' header has no efffect.
+    /// - Returns: A sequence of header values for `name`.
+    public func values(forHeader name: String, canonicalForm: Bool = false) -> Values {
+        if canonicalForm, name.isEqualCaseInsensitiveASCIIBytes(to: "set-cookie") {
+            // Not safe to split 'set-cookie'.
+            return Values(headers: self, name: name, canonicalize: false)
+        } else {
+            return Values(headers: self, name: name, canonicalize: canonicalForm)
+        }
+    }
+
+    /// Special internal function for use by tests.
+    internal subscript(position: Int) -> (String, String) {
```

Comment:
Can we put a leading underscore in the label?

Reply:
This isn't new API, GitHub did a bad job with the diff. Happy to make this change in a followup PR though.

Reply:
FWIW: there is no external label at the moment